index.js

```diff
@@ -5,19 +5,58 @@ import isDev from 'electron-is-dev';
 
 const isMacOS = process.platform === 'darwin';
 
-const developmentToolsOptions = {};
+// A Map allows each window to have its own options
+const developmentToolsOptions = new Map();
 
 function toggleDevelopmentTools(win = BrowserWindow.getFocusedWindow()) {
 	if (win) {
 		const {webContents} = win;
 		if (webContents.isDevToolsOpened()) {
 			webContents.closeDevTools();
 		} else {
-			webContents.openDevTools(developmentToolsOptions);
+			webContents.openDevTools(developmentToolsOptions.get(win));
 		}
 	}
 }
 
+function shouldRun(options) {
+	return options && (options.isEnabled === true || (options.isEnabled === null && isDev));
+}
+
+function getOptionsForWindow(win, options) {
+	if (!options.windowSelector) {
+		return options;
+	}
+
+	const newOptions = options.windowSelector(win);
+
+	return newOptions === true
+		? options
+		: (newOptions === false
+			? {isEnabled: false}
+			: {...options, ...newOptions});
+}
+
+function registerAccelerators(win = BrowserWindow.getFocusedWindow()) {
+	(async () => {
```

Comment:
This is moot. Just mark the function as `async` and then avoid awaiting it at the callsite.

Reply:
You are right, it's something I wanted to do in the refactor and I forgot.
Well pointed.